src/Apps/Order/Components/AddressVerificationFlow.tsx

```diff
@@ -195,90 +113,129 @@ const AddressVerificationFlow: React.FC<AddressVerificationFlowProps> = ({
-              lines: lines,
-              address: address,
-            }))
-          setAddressOptions([...suggestedOptions, inputOption])
-        } else {
-          setAddressOptions([inputOption])
-          setModalType(ModalType.REVIEW_AND_CONFIRM)
-          trackEvent({
-            action_type: "validationAddressViewed",
-            context_module: ContextModule.ordersShipping,
-            context_page_owner_type: OwnerType.ordersShipping,
-            context_page_owner_id: contextPageOwnerSlug,
-            user_id: userId,
-            flow: "user adding shipping address",
-            subject: "Check your delivery address",
-            option: "review and confirm",
-          })
-        }
+        addressOptions = [inputOption]
+        verificationPath = VerificationPath.REVIEW_AND_CONFIRM
       }
     }
+  }
+
+  const modalTitle = modalTitles[verificationPath]
+
+  const [selectedAddressKey, setSelectedAddressKey] = useState<
+    AddressOptionKey | undefined
+  >(addressOptions[0]?.key)
+
+  // perform only once on mount
+  useEffect(() => {
+    if (didLoad.current) return
+
+    if (verificationPath === VerificationPath.ERROR_IMMEDIATE_CONFIRM) {
+      const fallbackOption = fallbackFromFormValues(verificationInput)
+      onChosenAddress(AddressVerifiedBy.USER, fallbackOption.address)
+      didLoad.current = true
+    } else if (verificationPath === VerificationPath.IMMEDIATE_CONFIRM) {
+      onChosenAddress(AddressVerifiedBy.ARTSY, addressOptions[0].address!)
+      didLoad.current = true
+    } else {
+      setShowModal(true)
+      didLoad.current = true
+
+      trackViewedModal({
+        subject: modalTitle!,
+      })
+    }
   }, [
-    inputAddress,
-    onChosenAddress,
-    suggestedAddresses,
-    verificationStatus,
-    trackEvent,
-    userId,
-    contextPageOwnerSlug,
-    hasError,
+    didLoad,
+    addressOptions,
+    verificationPath,
     verificationInput,
+    onChosenAddress,
+    trackViewedModal,
+    modalTitle,
   ])
 
-  if (verificationStatus === "VERIFIED_NO_CHANGE" || hasError)
-    return <div data-testid="emptyAddressVerification"></div>
+  const chooseAddress = useCallback(() => {
+    if (!selectedAddressKey) return
+
+    const selectedAddress = addressOptions.find(
+      option => option.key === selectedAddressKey
+    )
 
-  if (!modalType || addressOptions.length === 0) return null
+    const verifiedBy =
+      selectedAddressKey === USER_INPUT
+        ? AddressVerifiedBy.USER
+        : AddressVerifiedBy.ARTSY
 
-  return (
+    if (selectedAddress) {
+      setShowModal(false)
+      onChosenAddress(verifiedBy, selectedAddress.address)
+    }
+  }, [addressOptions, onChosenAddress, selectedAddressKey])
+
+  const handleBackToEdit = ({
+    label,
+    option,
+    subject,
+  }: {
+    label: string
+    subject: string
+    option?: string
+  }) => {
+    trackClickedModal({
+      subject,
+      option,
+      label,
+    })
+    onClose()
+  }
```

Comment:
Realized that closing the modal is a different tracking event from choosing an address so i need to disambiguate this bit.